app/src/main/java/com/yuneec/example/component/fragment/TelemetryFragment.java

```diff
@@ -227,6 +234,80 @@ public void run() {
         }
     }
 
+    public class GPSInfoListener implements Telemetry.GPSInfoListener {
+
+        @Override
+        public void onGPSInfoCallback(Telemetry.GPSInfo gpsInfo) {
+            adapter.setItemValue(TelemetryIndices.GPS_NO_OF_SATELLITES, String.valueOf(gpsInfo.numSatellites));
+
+            getActivity().runOnUiThread(new Runnable() {
+                public void run() {
+                    adapter.notifyDataSetChanged();
+                }
+            });
+        }
+    }
+
+    public class HomePositionListener implements Telemetry.HomePositionListener {
+
+        @Override
+        public void onHomePositionCallback(Telemetry.Position position) {
+            adapter.setItemValue(TelemetryIndices.HOME_POSITION_LATITUDE, String.valueOf(position.latitudeDeg));
+            adapter.setItemValue(TelemetryIndices.HOME_POSITION_LONGITUDE,
+                                 String.valueOf(position.longitudeDeg));
+
+            getActivity().runOnUiThread(new Runnable() {
+                public void run() {
+                    adapter.notifyDataSetChanged();
+                }
+            });
+        }
+    }
+
+    public class ArmedListener implements Telemetry.ArmedListener {
+
+        @Override
+        public void onIsArmedCallback(boolean b) {
+            adapter.setItemValue(TelemetryIndices.IS_ARMED, String.valueOf(b));
+
+            getActivity().runOnUiThread(new Runnable() {
+                public void run() {
+                    adapter.notifyDataSetChanged();
+                }
+            });
+        }
+    }
+
+    public class InAirListener implements Telemetry.InAirListener {
+
+        @Override
+        public void onIsInAirCallback(boolean b) {
+            adapter.setItemValue(TelemetryIndices.IN_AIR, String.valueOf(b));
+
+            getActivity().runOnUiThread(new Runnable() {
+                public void run() {
+                    adapter.notifyDataSetChanged();
+                }
+            });
+        }
+    }
+
+    public class RCStatusListener implements Telemetry.RCStatusListener {
+
+        @Override
+        public void onRcStatusCallback(Telemetry.RCStatus rcStatus) {
+            adapter.setItemValue(TelemetryIndices.RC_STATUS,  String.format("%d",
+                                                                            (int)(100 * rcStatus.signalStrengthPercent)));
```

Comment:
This is actually pretty weird that it is called `Percent` when it is actually 0..1.

I should change that inside, your work is fine.